# Step 3: Create some Fake Data for DEV Purposes

This code below was originally just a script for making some interesting objects for the unattached context memory, including a DAG based on a promotable object -> Sighting, with sub objects. It does not use blocks to make the objects, and we just save the objects into the unattached list. So this workbook does not use the blocks, since this is a hack to make more data for the viz development

In [6]:

from stixorm.module.definitions.stix21 import (
    Identity, EmailAddress, UserAccount, Relationship, Bundle, Software, Process, File,
)
from stixorm.module.definitions.os_threat import (
    IdentityContact, EmailContact, SocialMediaContact, ContactNumber
)
from stixorm.module.definitions.stix21 import (
    ObservedData, IPv4Address, IPv6Address, EmailAddress, DomainName, EmailMessage, URL, UserAccount, File, MACAddress, Software, Process, NetworkTraffic, Artifact, Directory,
    Identity, Incident, Note, Sighting, Indicator, Relationship, Location, Software, Process, Bundle
)
from stixorm.module.definitions.os_threat import (
    StateChangeObject, EventCoreExt, Event, ImpactCoreExt,
    Availability, Confidentiality, External, Integrity, Monetary, Physical,
    Traceability, Impact, IncidentScoreObject, IncidentCoreExt, TaskCoreExt,
    Task, SightingEvidence, Sequence, SequenceExt, ContactNumber, EmailContact,
    SocialMediaContact, IdentityContact, AnecdoteExt, Anecdote,
    SightingAnecdote, SightingAlert, SightingContext, SightingExclusion,
    SightingEnrichment, SightingHunt, SightingFramework, SightingExternal
)
import sys
sys.path.append('../')
import os
from Block_Families.General._library.convert_n_and_e import convert_node, convert_relns, convert_sighting
from Block_Families.OS_Triage.Viz_Dataviews.get_unattached import get_unattached
from Utilities.local_make_general import invoke_save_company_context_block, invoke_save_incident_context_block
import json
import os
from Utilities.util import conv
# 1. Variables
results_base = "../Orchestration/Results/"
examples_base = "../Orchestration/Results/Examples/"
context_base = "../Orchestration/generated/os-triage/context_mem/"
context_map = "context_map.json"
scratch_name = "scratch.json"
scratch = {}
final_nodes = []
final_ids = []
final_edges = []
s_nodes = []
s_edges = []
r_nodes = []
r_edges = []
normal = []
relns = []
unattached = []
edges = []
relation_edges = []
relation_replacement_edges = []
relations = []
#
incident_data = {
    "incident" : "/incident.json",
    "start" : "/sequence_start_refs.json",
    "sequence" : "/sequence_refs.json",
    "impact" : "/impact_refs.json",
    "event" : "/event_refs.json",
    "task" : "/task_refs.json",
    "other" : "/other_object_refs.json",
    "unattached" : "/unattached_objs.json",
    "relations" : "/incident_relations.json",
    "edges" : "/incident_edges.json",
    "relation_edges" : "/relation_edges.json",
    "relation_replacement_edges" : "/relation_replacement_edges.json"
}
context_type = {
    "context_type": "unattached"
}
# ident_results_obj_path = results_base + "/step0/Usr1__ident.json"
# ident_results_context_path = results_base  + "/step0/context/usr_1__ident_context.json"
# result = invoke_save_incident_context_block(ident_results_obj_path, ident_results_context_path, context_type)
# print(f" result->{result}")
# UNATTACHED STUFF
# A. Software
software = Software(name="evil.exe")
normal.append(software)
s_nodes, s_edges = convert_node(conv(software))
# B. File
hashes ={"SHA-256": "fe90a7e910cb3a4739bed9180e807e93fa70c90f25a8915476f5e4bfbac681db"}
file = File(name="evil.exe", hashes=hashes)
normal.append(file)
nodes2, edges2 = convert_node(conv(file))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
print(f'file edges->', s_edges)
# C. Process
process = Process(pid=1221, created_time="2023-01-20T14:11:25.55Z",
                  command_line="./gedit-bin --destroy-alll", image_ref=file.id)
nodes2, edges2 = convert_node(conv(process))
normal.append(process)
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
print(f'edges2 edges->', edges2)
print(f'process edges->', s_edges)


# A. Email Addresses and User Accounts
user_account3 = UserAccount(account_type="unix", account_login="sthor", display_name="silly thor")
normal.append((user_account3))
nodes2, edges2 = convert_node(conv(user_account3))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
email_addr3 = EmailAddress(value="silly@mycompany.com", belongs_to_ref=user_account3.id)
normal.append(email_addr3)
nodes2, edges2 = convert_node(conv(email_addr3))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
user_account4 = UserAccount(account_type="unix", account_login="sguy", display_name="strange guy")
normal.append(user_account4)
nodes2, edges2 = convert_node(conv(user_account4))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
email_addr4 = EmailAddress(value="strange@mycompany.com", belongs_to_ref=user_account4.id)
nodes2, edges2 = convert_node(conv(email_addr4))
normal.append(email_addr4)
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
user_account5 = UserAccount(account_type="unix", account_login="dguy", display_name="dumbo guy")
normal.append(user_account5)
nodes2, edges2 = convert_node(conv(user_account5))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
email_addr5 = EmailAddress(value="dumbo@mycompany.com", belongs_to_ref=user_account5.id)
normal.append(email_addr5)
nodes2, edges2 = convert_node(conv(email_addr5))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
# B. Duplicate-of SRO
sro4 = Relationship(relationship_type="duplicate-of", source_ref=email_addr3.id, target_ref=email_addr4)
relns.append(sro4)
nodes2, embedded, edges2, relation_replacement_edges = convert_relns(conv(sro4))
r_nodes = r_nodes + nodes2
r_edges = r_edges + edges2
print(f'\nembedded edges->', embedded)
print(f'\nedges2 edges->', edges2)
print(f'\nrelation_replacement_edges edges->', relation_replacement_edges)
sro5 = Relationship(relationship_type="duplicate-of", source_ref=email_addr3.id, target_ref=email_addr5)
relns.append(sro5)
nodes2, embedded, edges2, relation_replacement_edges = convert_relns(conv(sro5))
r_nodes = r_nodes + nodes2
r_edges = r_edges + edges2
email_message1 = EmailMessage(to_refs=email_addr3.id, from_ref=email_addr4.id, subject="we are coming for you",
                              is_multipart=False, date="2020-10-19T01:01:01.000Z")
nodes2, edges2 = convert_node(conv(email_message1))
normal.append(email_message1)
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
url1 = URL(value="https://www.northkorea.nk/we/are/mad/")
nodes2, edges2 = convert_node(conv(url1))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
rel1 = Relationship(relationship_type='derived-from', source_ref=email_message1.id, target_ref=url1.id)
relns.append(rel1)
nodes2, embedded, edges2, relation_replacement_edges = convert_relns(conv(rel1))
r_nodes = r_nodes + nodes2
r_edges = r_edges + edges2
# C. Setup observed-data
obs_refs2 = [email_addr3.id, user_account3.id, url1.id, rel1.id, email_message1.id,
             email_addr4.id, user_account4.id, email_addr5.id, user_account5.id,
             sro4.id, sro5.id]
observation3 = ObservedData(number_observed=1, object_refs=obs_refs2,
                            first_observed=email_message1.date,last_observed=email_message1.date)
normal.append(observation3)
nodes2, edges2 = convert_node(conv(observation3))
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
#
# D. Setup Exchange Server Identity
exchange = Identity(name="Microsoft Exchange", description="Microsoft Exchange Server",
                    identity_class="system")
nodes2, edges2 = convert_node(conv(exchange))
normal.append(exchange)
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
pat1 = "[email-addr:value = '" + email_addr3.value + "' AND email:subject = '" + email_message1.subject + "']"
ind1 = Indicator(name="Suspicious Email", pattern_type="stix", pattern=pat1, indicator_types=["unknown"])
nodes2, edges2 = convert_node(conv(ind1))
normal.append(ind1)
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2
#
# E. Setup Sighting with Context Evidence
query = "query from:"+email_addr3.value+", subject:"+email_message1.subject
value = "["+email_addr3.value+", "+email_addr4.value+", "+email_addr5.value+"]"
context = SightingContext(name="Exchange", description=query, value=value)
sight_ext = SightingEvidence(extension_type="property-extension")
sight_ext_id = "extension-definition--0d76d6d9-16ca-43fd-bd41-4f800ba8fc43"
sight_context_ext = {
    sight_ext_id: sight_ext,
    "sighting-context": context
}
sighting3 = Sighting(observed_data_refs=observation3.id,
                     sighting_of_ref=ind1.id, extensions=sight_context_ext)
nodes2, edges2 = convert_sighting(conv(sighting3))
normal.append(sighting3)
print("\n###############################################")
print("nodes2->", nodes2, "\n\n")
print("edges2->", edges2)
s_nodes = s_nodes + nodes2
s_edges = s_edges + edges2


scratch['nodes'] = s_nodes + r_nodes
scratch['edges'] = s_edges + r_edges
with open(examples_base + scratch_name, "w") as outfile:
    json.dump(scratch, outfile)


file edges-> []
edges2 edges-> [{'name': 'image-of', 'type': 'embedded', 'source': 'file--e7c7ddeb-d693-53e7-bcd8-f40951888120', 'target': 'process--b1d38696-caa9-425e-a501-a38b4a1e95b5', 'id': 'file--e7c7ddeb-d693-53e7-bcd8-f40951888120-process--b1d38696-caa9-425e-a501-a38b4a1e95b5'}]
process edges-> [{'name': 'image-of', 'type': 'embedded', 'source': 'file--e7c7ddeb-d693-53e7-bcd8-f40951888120', 'target': 'process--b1d38696-caa9-425e-a501-a38b4a1e95b5', 'id': 'file--e7c7ddeb-d693-53e7-bcd8-f40951888120-process--b1d38696-caa9-425e-a501-a38b4a1e95b5'}]

embedded edges-> []

edges2 edges-> [{'stix-id': 'relationship--ea056bcc-bb4d-42ee-9f28-17c15c9c5689', 'type': 'relationship', 'name': 'duplicate-of', 'source': 'email-addr--a7155ffe-8ea7-563a-a54c-e94e23ec50c8', 'target': 'relationship--ea056bcc-bb4d-42ee-9f28-17c15c9c5689', 'id': 'email-addr--a7155ffe-8ea7-563a-a54c-e94e23ec50c8-relationship--ea056bcc-bb4d-42ee-9f28-17c15c9c5689'}, {'stix-id': 'relationship--ea056bcc-bb4d-42ee-9f28-17

In [7]:

def add_node(node, context_dir, context_type):
    print(f"node->{node}")
    print(f"context_dir->{context_dir}")
    print(f"context type->{context_type}")
    print(f"path->{context_dir + incident_data[context_type]}")
    exists = False
    stix_nodes_list = []
    if  os.path.exists(context_dir + incident_data[context_type]):
        with open(context_dir + incident_data[context_type], "r") as mem_input:
            stix_nodes_list = json.load(mem_input)
            for i in range(len(stix_nodes_list)):
                if stix_nodes_list[i]["id"] == node["id"]:
                    stix_nodes_list[i] = node
                    exists = True
            if not exists:
                stix_nodes_list.append(node)
    else:
        stix_nodes_list = [node]
    with open(context_dir + incident_data[context_type], 'w') as f:
        f.write(json.dumps(stix_nodes_list))


def add_edge(edge, context_dir, context_type):
    exists = False
    stix_edge_list = []
    if os.path.exists(context_dir + incident_data[context_type]):
        with open(context_dir + incident_data[context_type], "r") as mem_input:
            stix_edge_list = json.load(mem_input)
            for i in range(len(stix_edge_list)):
                if stix_edge_list[i]["source"] == edge["source"] and stix_edge_list[i]["target"] == edge["target"]:
                    stix_edge_list[i] = edge
                    exists = True
            if not exists:
                stix_edge_list.append(edge)
    else:
        stix_edge_list = [edge]
    with open(context_dir + incident_data[context_type], 'w') as f:
        f.write(json.dumps(stix_edge_list))

context_type = "unattached"
local_map = {}
with open(context_base + context_map, "r") as current_context:
    local_map = json.load(current_context)
    current_incident_dir = local_map["current_incident"]
    TR_Incident_Context_Dir = context_base + current_incident_dir
    # 1. Extract the components of the object
    TR_Context_Incident = TR_Incident_Context_Dir + incident_data["incident"]
    TR_Context_Filename = TR_Context_Incident + incident_data[context_type]

    for norm in normal:
        cnorm = conv(norm)
        if cnorm["type"] == "sighting":
            nodes, edges = convert_sighting(cnorm)
            add_node(nodes[0], TR_Incident_Context_Dir, context_type)
            for edge in edges:
                add_edge(edge, TR_Incident_Context_Dir, "edges")
        else:
            nodes, edges = convert_node(cnorm)
            print(f"nodes->{nodes}")
            print(f'type of->{type(nodes[0])}')
            print(f"nodes[0]->{nodes[0]}")
            add_node(nodes[0], TR_Incident_Context_Dir, context_type)
            for edge in edges:
                add_edge(edge, TR_Incident_Context_Dir, "edges")

    for reln in relns:
        creln = conv(reln)
        nodes, edges, relation_edges, relation_replacement_edges = convert_relns(creln)
        add_node(nodes[0],TR_Incident_Context_Dir, "relations")
        for edge in edges:
            add_edge(edge, TR_Incident_Context_Dir, "edges")
        for edge in relation_edges:
            add_edge(edge, TR_Incident_Context_Dir, "relation_edges")
        for edge in relation_replacement_edges:
            add_edge(edge, TR_Incident_Context_Dir, "relation_replacement_edges")


nodes->[{'id': 'software--3974613c-1980-5209-8f08-17920a04da04', 'type': 'software', 'original': {'type': 'software', 'spec_version': '2.1', 'id': 'software--3974613c-1980-5209-8f08-17920a04da04', 'name': 'evil.exe'}, 'icon': 'software', 'name': 'Software', 'heading': 'Software -> evil.exe', 'description': '', 'object_form': 'software', 'object_group': 'sco-forms', 'object_family': 'stix-forms'}]
type of-><class 'dict'>
nodes[0]->{'id': 'software--3974613c-1980-5209-8f08-17920a04da04', 'type': 'software', 'original': {'type': 'software', 'spec_version': '2.1', 'id': 'software--3974613c-1980-5209-8f08-17920a04da04', 'name': 'evil.exe'}, 'icon': 'software', 'name': 'Software', 'heading': 'Software -> evil.exe', 'description': '', 'object_form': 'software', 'object_group': 'sco-forms', 'object_family': 'stix-forms'}
node->{'id': 'software--3974613c-1980-5209-8f08-17920a04da04', 'type': 'software', 'original': {'type': 'software', 'spec_version': '2.1', 'id': 'software--3974613c-1980-5209-

## Handy raw SCO stuff, from OCA Github

In [8]:
data = [

    {
        "type": "ipv4-addr",
        "value": "13.14.1.1"
    },
    {
        "type": "url",
        "value": "http://maliciousmalware.bad"
    },
    {
        "type": "ipv4-addr",
        "value": "9.9.9.9"
    },
    {
        "type": "mac-addr",
        "value": "00-00-11-00-22-00"
    },
    {
        "type": "ipv4-addr",
        "value": "192.168.xxx.yyy"
    },
            {
                "type": "mac-addr",
                "value": "12:34:56:78:9a:bc"
            },
            {
                "type": "ipv4-addr",
                "value": "192.0.2.0"
            },
            {
                "type": "artifact",
                "payload_bin": "[original event log data base 64 encoded]"
            },
            {
                "type": "user-account",
                "user_id": "S-1-2-3-4",
                "account_login": "joe@example.com"
            },
            {
                "path": "C:\\Program Files\\Google\\Chrome\\Application",
                "type": "directory"
            },
            {
                "type": "url",
                "value": "https://example.com/download"
            },
            {
                "type": "ipv6-addr",
                "value": "2001:db8::"
            },
            {
                "type": "domain-name",
                "value": "example.com"
            }
]
data2 =   {
    "type": "ipv6-addr",
    "value": "ffff:1111:ffff:1111:ffff:1111:ffff:1111",
    "resolves_to_refs": [
      "7"
    ]
  }

data3 = {
                "type": "x-oca-geo",
                "country_iso_code": "US"
        }
data4 = [

            {
                "name": "chrome.exe",
                "type": "file",
                "hashes":
                {
                    "SHA-1": "114c52257780067e3f8bfab4d2706be6debc0ace"
                },
                "parent_directory_ref": "7"
            },
            {
                "pid": 14068,
                "name": "chrome.exe",
                "type": "process",
                "created": "2022-07-15T15:41:11.3214492Z",
                "command_line": "C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe --disable-features --enable-features",
                "creator_user_ref": "5",
                "parent_ref": "9",
                "binary_ref": "6"
            },
            {
                "pid": 7116,
                "name": "explorer.exe",
                "type": "process",
                "created": "2022-07-15T08:01:20.813Z",
                "creator_user_ref": "5"
            },
            {
                "type": "network-traffic",
                "src_ref": "3",
                "dst_ref": "12",
                "src_port": 2041,
                "dst_port": 443,
                "protocols": ["tcp", "https"]
            }
]
mac = {}
local_list = []
for dat in data:
    if dat["type"] == 'url':
        local_object = URL(**dat)
        print(conv(local_object))
    elif dat["type"] == "software":
        local_object = Software(**dat)
        print(conv(local_object))
    elif dat["type"] == "process":
        local_object = Process(**dat)
        print(conv(local_object))
    elif dat["type"] == "artifact":
        pass
        # local_object = Artifact(**dat)
        # print(conv(local_object))
    elif dat["type"] == "user-account":
        local_object = UserAccount(**dat)
        print(conv(local_object))
    elif dat["type"] == "domain-name":
        local_object = DomainName(**dat)
        print(conv(local_object))
    elif dat["type"] == "network-traffic":
        local_object = NetworkTraffic(**dat)
        print(conv(local_object))
    elif dat["type"] == "ipv6-addr":
        local_object = IPv6Address(**dat)
        print(conv(local_object))
    elif dat["type"] == 'ipv4-addr':
        local_object = IPv4Address(**dat)
        print(conv(local_object))
    elif dat["type"] == 'mac-addr':
        mac = MACAddress(**dat)
        print(conv(mac))
        # data2["resolves_to_refs"] = local_list.append(mac["id"])
        # print(data2)
        #



{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--157aeca1-b949-5c83-89cb-ef99757be9c6', 'value': '13.14.1.1'}
{'type': 'url', 'spec_version': '2.1', 'id': 'url--e9653306-7457-5316-a478-57559c06d87f', 'value': 'http://maliciousmalware.bad'}
{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--38609426-bc06-5cbc-9763-e77e99c9d9e4', 'value': '9.9.9.9'}
{'type': 'mac-addr', 'spec_version': '2.1', 'id': 'mac-addr--b5dbdbbb-5470-58c8-9b86-81b7d83a770b', 'value': '00-00-11-00-22-00'}
{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--f1958dd5-5175-5702-beef-742d2cf3c0ec', 'value': '192.168.xxx.yyy'}
{'type': 'mac-addr', 'spec_version': '2.1', 'id': 'mac-addr--c531c16b-cb43-51c7-89f4-6d01507dd6e5', 'value': '12:34:56:78:9a:bc'}
{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--d1f1b1e2-d0a9-5fca-bc1b-adae31c22ad3', 'value': '192.0.2.0'}
{'type': 'user-account', 'spec_version': '2.1', 'id': 'user-account--1a9c7df9-1266-5a73-8d67-ceeb4f9d8fd8', 'u

In [9]:
first_dir = Directory(**{
    "path": "C:\\Program Files\\Google\\Chrome\\Application",
    "type": "directory"
  })
print(f"first-dir", first_dir)
second_dir = Directory(**{
    "type": "directory",
    "path": "C:\\Users\\joe\\Downloads"
  })
print(f"2nd-dir", second_dir)
file1 = {
    "type": "file",
    "name": "example.exe",
    "parent_directory_ref": second_dir["id"],
    "hashes": {
      "SHA-1": "223c52257780067e3f8bfab4d2706be6debc0aca"
    }
  }
f1 = File(**file1)
print(f"f1 ", f1)
file2 =  {
    "name": "chrome.exe",
    "type": "file",
    "hashes": {
      "SHA-1": "114c52257780067e3f8bfab4d2706be6debc0ace"
    },
    "parent_directory_ref": first_dir["id"]
  }
f2 = File(**file2)
print(f"f2 ", f2)
process2=  {
    "pid": 7116,
    "type": "process",
    "creator_user_ref": "user-account--1a9c7df9-1266-5a73-8d67-ceeb4f9d8fd8"
  }
p2 = Process(**process2)
print(f"p2 ", p2)
process3 = Process(**{
    "pid": 14068,
    "type": "process",
    "command_line": "C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe --disable-features --enable-features",
    "creator_user_ref": "user-account--1a9c7df9-1266-5a73-8d67-ceeb4f9d8fd8",
    "parent_ref": p2["id"],
    "image_ref": f2["id"]
  })

p3 = Process(**process3)
print(f"p3 ", p3)
print(f"process3", process3)
{"created": "2022-07-15T15:41:11.3214492Z",
    "name": "explorer.exe",
    "name": "chrome.exe" }

first-dir {"type": "directory", "spec_version": "2.1", "id": "directory--325be77f-a8bf-5259-8a1d-dbf5984f0134", "path": "C:\\Program Files\\Google\\Chrome\\Application"}
2nd-dir {"type": "directory", "spec_version": "2.1", "id": "directory--cde02145-c164-53f0-ab5a-681e313342e2", "path": "C:\\Users\\joe\\Downloads"}
f1  {"type": "file", "spec_version": "2.1", "id": "file--1bd39816-c38f-5d1c-a768-0420c397b79d", "hashes": {"SHA-1": "223c52257780067e3f8bfab4d2706be6debc0aca"}, "name": "example.exe", "parent_directory_ref": "directory--cde02145-c164-53f0-ab5a-681e313342e2"}
f2  {"type": "file", "spec_version": "2.1", "id": "file--7fe91bca-bbdc-5d55-aa55-7fd2204cb2f6", "hashes": {"SHA-1": "114c52257780067e3f8bfab4d2706be6debc0ace"}, "name": "chrome.exe", "parent_directory_ref": "directory--325be77f-a8bf-5259-8a1d-dbf5984f0134"}
p2  {"type": "process", "spec_version": "2.1", "id": "process--448b57ad-12d0-4936-9b8a-c4a68993f7b8", "pid": 7116, "creator_user_ref": "user-account--1a9c7df9-1266-5a

{'created': '2022-07-15T15:41:11.3214492Z', 'name': 'chrome.exe'}

In [10]:
event1 = [
    {
                "type": "mac-addr",
                "value": "12:34:56:78:9a:bc"
            },
            {
                "type": "ipv4-addr",
                "value": "192.0.2.0"
            },
    {
                "path": "C:\\Program Files\\Google\\Chrome\\Application",
                "type": "directory"
            },
            {
                "type": "user-account",
                "user_id": "S-1-2-3-4",
                "account_login": "joe@example.com"
            },
            {
                "path": "C:\\Program Files\\Google\\Chrome\\Application",
                "type": "directory"
            },
            {
                "type": "url",
                "value": "https://example.com/download"
            },
            {
                "type": "ipv6-addr",
                "value": "2001:db8::"
            },
            {
                "type": "domain-name",
                "value": "example.com"
            }
]
mac = {}
local_list = []
for dat in event1:
    if dat["type"] == 'url':
        local_object = URL(**dat)
        print(conv(local_object))
    elif dat["type"] == "software":
        local_object = Software(**dat)
        print(conv(local_object))
    elif dat["type"] == "process":
        local_object = Process(**dat)
        print(conv(local_object))
    elif dat["type"] == "directory":
        local_object = Directory(**dat)
        print(conv(local_object))
    elif dat["type"] == "artifact":
        pass
        # local_object = Artifact(**dat)
        # print(conv(local_object))
    elif dat["type"] == "user-account":
        local_object = UserAccount(**dat)
        print(conv(local_object))
    elif dat["type"] == "domain-name":
        local_object = DomainName(**dat)
        print(conv(local_object))
    elif dat["type"] == "network-traffic":
        local_object = NetworkTraffic(**dat)
        print(conv(local_object))
    elif dat["type"] == "ipv6-addr":
        local_object = IPv6Address(**dat)
        print(conv(local_object))
    elif dat["type"] == 'ipv4-addr':
        local_object = IPv4Address(**dat)
        print(conv(local_object))
    elif dat["type"] == 'mac-addr':
        mac = MACAddress(**dat)
        print(conv(mac))

net1 = NetworkTraffic(**{
                "type": "network-traffic",
                "src_ref": "ipv4-addr--d1f1b1e2-d0a9-5fca-bc1b-adae31c22ad3",
                "dst_ref": "ipv6-addr--7b18f904-dbfb-5c92-8b83-c71f1832960f",
                "src_port": 2041,
                "dst_port": 443,
                "protocols": ["tcp", "https"]
            })
print(conv(net1))
explorer =  Process(**{
                "pid": 7116,
                "type": "process",
                "creator_user_ref": "user-account--1a9c7df9-1266-5a73-8d67-ceeb4f9d8fd8"
            })
print(conv(explorer))
chrome2 =   File(**{
                "name": "chrome.exe",
                "type": "file",
                "hashes":
                {
                    "SHA-1": "114c52257780067e3f8bfab4d2706be6debc0ace"
                },
                "parent_directory_ref": "directory--325be77f-a8bf-5259-8a1d-dbf5984f0134"
            })
print(conv(chrome2))


chrome = Process(**{
                "pid": 14068,
                "type": "process",
                "command_line": "C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe --disable-features --enable-features",
                "creator_user_ref": "user-account--1a9c7df9-1266-5a73-8d67-ceeb4f9d8fd8",
                "parent_ref": explorer["id"],
                "image_ref": chrome2["id"]
            })
print(conv(chrome))


{'type': 'mac-addr', 'spec_version': '2.1', 'id': 'mac-addr--c531c16b-cb43-51c7-89f4-6d01507dd6e5', 'value': '12:34:56:78:9a:bc'}
{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--d1f1b1e2-d0a9-5fca-bc1b-adae31c22ad3', 'value': '192.0.2.0'}
{'type': 'directory', 'spec_version': '2.1', 'id': 'directory--325be77f-a8bf-5259-8a1d-dbf5984f0134', 'path': 'C:\\Program Files\\Google\\Chrome\\Application'}
{'type': 'user-account', 'spec_version': '2.1', 'id': 'user-account--1a9c7df9-1266-5a73-8d67-ceeb4f9d8fd8', 'user_id': 'S-1-2-3-4', 'account_login': 'joe@example.com'}
{'type': 'directory', 'spec_version': '2.1', 'id': 'directory--325be77f-a8bf-5259-8a1d-dbf5984f0134', 'path': 'C:\\Program Files\\Google\\Chrome\\Application'}
{'type': 'url', 'spec_version': '2.1', 'id': 'url--de785076-713c-567d-ae63-b646ec3f9374', 'value': 'https://example.com/download'}
{'type': 'ipv6-addr', 'spec_version': '2.1', 'id': 'ipv6-addr--7b18f904-dbfb-5c92-8b83-c71f1832960f', 'value': '2001:db8::'}
{'

{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--157aeca1-b949-5c83-89cb-ef99757be9c6', 'value': '13.14.1.1'},
{'type': 'url', 'spec_version': '2.1', 'id': 'url--e9653306-7457-5316-a478-57559c06d87f', 'value': 'http://maliciousmalware.bad'},
{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--38609426-bc06-5cbc-9763-e77e99c9d9e4', 'value': '9.9.9.9'},
{'type': 'mac-addr', 'spec_version': '2.1', 'id': 'mac-addr--b5dbdbbb-5470-58c8-9b86-81b7d83a770b', 'value': '00-00-11-00-22-00'},
{'type': 'ipv4-addr', 'spec_version': '2.1', 'id': 'ipv4-addr--f1958dd5-5175-5702-beef-742d2cf3c0ec', 'value': '192.168.xxx.yyy'},
{'type': 'ipv6-addr', 'spec_version': '2.1', 'id': 'ipv6-addr--dc0ef52f-b6cf-52a0-a6a1-977b32a57063', 'value': 'ffff:1111:ffff:1111:ffff:1111:ffff:1111'}
